### ISSUE TITLE
CUDA compilation error after  #3240

### ISSUE THREAD

**Issue by JasonRuonanWang** [ornladios/ADIOS2#3244]
/home/jason/d/a2/source/adios2/helper/adiosMemory.cpp: In function ‘int adios2::helper::NdCopy(const char*, const adios2::helper::CoreDims&, const adios2::helper::CoreDims&, bool, bool, char*, const adios2::helper::CoreDims&, const adios2::helper::CoreDims&, bool, bool, int, const adios2::helper::CoreDims&, const adios2::helper::CoreDims&, const adios2::helper::CoreDims&, const adios2::helper::CoreDims&, bool, adios2::MemorySpace)’:
/home/jason/d/a2/source/adios2/helper/adiosMemory.cpp:459:61: error: invalid initialization of reference of type ‘adios2::Dims&’ {aka ‘std::vector<long unsigned int>&’} from expression of type ‘adios2::helper::DimsArray’
  459 |                 helper::NdCopyCUDA(inOvlpBase, outOvlpBase, inOvlpGapSize,
      |                                                             ^~~~~~~~~~~~~
In file included from /home/jason/d/a2/source/adios2/helper/adiosMemory.h:285,
                 from /home/jason/d/a2/source/adios2/helper/adiosMemory.cpp:11:
/home/jason/d/a2/source/adios2/helper/adiosMemory.inl:103:37: note: in passing argument 3 of ‘void adios2::helper::NdCopyCUDA(const char*&, char*&, adios2::Dims&, adios2::Dims&, adios2::Dims&, std::size_t, std::size_t)’
  103 |                               Dims &inOvlpGapSize, Dims &outOvlpGapSize,
      |                               ~~~~~~^~~~~~~~~~~~~
make[2]: *** [source/adios2/CMakeFiles/adios2_core.dir/build.make:370: source/adios2/CMakeFiles/adios2_core.dir/helper/adiosMemory.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:4031: source/adios2/CMakeFiles/adios2_core.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
make -j install  1.89s user 0.48s system 146% cpu 1.628 total

**Comment by eisenhauer** [comment]
Ugh, I forgot that we still don't have CUDA in CI...  Will do.

**Comment by JasonRuonanWang** [comment]
Yeah, we really need to add CUDA in CI. Otherwise all the GPU related stuff will never be production.
